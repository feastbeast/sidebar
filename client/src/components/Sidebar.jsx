import React from 'react';
import ReactDOM from 'react-dom'
import Menu from './Menu.jsx';
import Booking from './Booking.jsx';
import Hours from './Hours.jsx';
import Contact from './Contact.jsx';
import GMap from './GMap.jsx';
const axios = require('axios');
//import '../../dist/sidebar.css';


export default class Sidebar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      place: {},
      isLoaded: false
    };

    this.getPlace = this.getPlace.bind(this);
  };
  componentDidMount() {
    if (typeof(window) !== 'undefined') {
      this.getPlace();
    }
  };

  getPlace() {
    let context = this;
    const id = this.props.restaurantId;
    //let id = window.location.href.split('/')[4];
    axios.get(`${BASE_URL}/api/restaurants/${id}`)
    .then((res) => {
      this.setState({
        place: res.data,
        isLoaded: true
      })
    })
    .catch((err) => console.log(err));
  };

  render() {
    const { hours, address, phone, website, location, id, name, url, menu_url, coordslat, coordslng } = this.state.place;
    if (!this.state.isLoaded) {
      return <div>Loading...</div>
    }
    if (this.state.isLoaded) {
      return (
        
          <div className="sidebar">
            <Menu  menuUrl={menu_url}/>
            <div className="greyBar"></div>
            <Booking />
            <div className="greyBar"></div>
            <div className="inSidebar">
              <Hours hours={hours}/>
              <Contact
                address={address} 
                phone={phone}
                website={url}
                lat={coordslat}
                lng={coordslng}
                id={id}
                name={name}
              />
              <GMap 
                location={location}
                id={id}
                lat={coordslat}
                lng={coordslng}
              />
            </div>
          </div>
   
      );
    }
  }
}


