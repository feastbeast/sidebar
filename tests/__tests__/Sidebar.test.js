import React from 'react';
import Enzyme from 'enzyme';
import {shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Sidebar from '../../client/src/components/Sidebar';

Enzyme.configure({ adapter: new Adapter() });

describe('Sidebar', () => {
  it('should be defined', () => {
    expect(Sidebar).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <Sidebar />
    );
    expect(tree).toMatchSnapshot();
  });

});
