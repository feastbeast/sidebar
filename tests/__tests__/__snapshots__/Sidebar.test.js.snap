// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<div>
  Loading...
</div>
`;
